feat: adds github-dark-vscode-theme

Justintime50 opened the pull request:
### Brief summary of what the package does

The [GitHub Dark Theme](https://marketplace.visualstudio.com/items?itemName=GitHub.github-vscode-theme) from Visual Studio Code ported to Emacs.

### Direct link to the package repository

https://github.com/Justintime50/github-dark-vscode-emacs-theme

### Your association with the package

Package author

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly (**does, however, per other themes I've elected to not byte compile**)
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  A couple quick suggestions:

- github-dark-vscode-theme.el#L24: Single-line comments should usually begin with `;;`
- For simple packages we usually recommend using the default recipe if your repository can be (re)structured to handle it, since it gives you the option to grow the package without further PRs if that ever comes to be
- Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to github-dark-vscode-theme.el
- I suggest expanding the summary line at the top of your file from "Emacs Theme" to something a bit more descriptive; any keywords you include here will help with discoverability

And just as an fyi, there are at least [a couple](https://melpa.org/#/?q=vs%20dark#/?q=vs%20dark#/?q=vs%20dark) themes with similar names, but they don't seem to use the same color palette, and I guess are based around the Dark+ theme instead of the Dark theme (I'm not familiar).

Justintime50 commented:
@riscy thanks for taking a look!

1. Fixed the comments
2. "Default recipe" meaning a flatter repo? Such that I don't need to include the `files` bit since it'll live in the root of the project? Not a huge fan of mixing non-code files with `src` files in my repos but happy to do it if it's the norm via MELPA.
3. Added license identifier
4. Corrected the summary line and commentary

Yeah, there are a few similar themes; however, this one does differ from the others and it is for a separate theme.

Let me know if there's anything else you'd like to get this merged.

riscy commented:
Happy to merge this now, but I'll leave a couple more comments below.

> meaning a flatter repo?

I see you already accommodated this one, and thanks a lot for that.  MELPA can accommodate all kinds of repo structures obviously, but we always give a gentle push toward the default recipe because it means adding files doesn't require another  MELPA PR.

> Added license identifier

Thanks!  I would still recommend an [SPDX-License-Identifier](https://spdx.dev/ids/) or official license boilerplate since these are a little more formal (and hence machine detectable with much more confidence), but this doesn't block.

Justintime50 commented:
Thanks for merging this! Means a lot.

Is there anything else needed to get this package to show up in the package list? I've run `M-x package-refresh-contents` and am still unable to see it on the package list.
